gantry/simulator/CMakeLists.txt

```diff
@@ -1,18 +1,67 @@
+# Load free for the posix port
+find_package(FreeRTOS)
+
 # Set up the include directories
 include_directories(
         .
+        ../firmware
         ${CMAKE_CURRENT_SOURCE_DIR}/../../include
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/include
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils
+
+)
 
+# Create the FreeRTOS source file list
+FILE(
+        GLOB FREERTOS_SOURCES
+        ${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/*.c
 )
 
+# Add the posix port files
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/MemMang/heap_3.c")
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/utils/wait_for_event.c")
+list(APPEND FREERTOS_SOURCES "${FreeRTOS_SOURCE_DIR}/FreeRTOS/Source/portable/ThirdParty/GCC/Posix/port.c")
+
+
+# Create the freertos lib
+add_library(freertos-gantry STATIC ${FREERTOS_SOURCES})
+
+if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Linux")
+    add_compile_definitions(USE_SOCKETCAN)
```

Comment:
We shouldn't use project-wide cmake commands. We should instead use target-specific commands (the equivalent here would be [target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html) on a target. If we need to add it to multiple targets, then we should.